### PR TITLE
bump crengine: add support for <ruby> elements

### PR DESCRIPTION
**Pull request by poire-z**
Includes https://github.com/koreader/crengine/pull/347 :
- Fix some non-recursive subtree walkers
- Text: skip formatting optimisations when inlineBoxes
- CSS parsing: parse selectors starting with [attrib]
- Use T="" for internal elements attributes
- CSS: re-order css_d_* (display) enum for easier checking
- Rendering methods: remove erm_table_cell
- Rendering methods: remove erm_table_caption
- Rendering methods: remove erm_list_item
- getRenderedWidths(): better estimate table width
- styleToTextFmtFlags(): adds is_block parameter
- Adds support for "display: ruby" and `<ruby>` elements
- Fix prev/next-VisibleWord-Start/End() with CJK text

Also:
- bump HarfBuzz to 2.6.8 https://github.com/koreader/koreader-base/pull/1122
- bump evernote-lua-sdk https://github.com/koreader/koreader-base/pull/1121

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/6305)

### CODE REVIEW

**Comment by poire-z** [comment]
I updated the book style tweak sample to use `0.8rem` instead of `80%` because that's what I've been happily using for some time, and it's easily changeable there if it's not the best option.

I'm not sure if I should do the same for the 3 other "In-page footnotes (small)" variants, as it happened to me that I had books with `body { font-size: 0.8em or 1.2em}` which had me increase or decrease KOReader's font size to get my usual prefered font size - using 80% would work in that case, while using 0.8rem would make the footnotes font too large.

**Comment by poire-z** [comment]
> my quick ruby in the margin demo could be included

It's a bit tedious to get right and not ugly :)
With your absolute sizes in px, I would get on the emulator:
<kbd>image</kbd>
and one would need to finely adjust Zoom/DPI to get something somehow OK (and bad if you go larger than OK).

And if there are 2 ruby in the same line, the 2 annotations would override each other:
<kbd>image</kbd>

So, I'd rather let that to the user via User style tweaks - so he can try to solve/size it according to his setup.

Also, I'd rather not advocate the use of ruby for wordwise stuff, because soon, someone will see the extra space added when the definition is longer than the base word:
<kbd>image</kbd>
and will ask for this to be fixed - and that won't really be as easily fixable as implementing this ruby support with inline-table (because that would mean sizing the inline-table only with the main word, and find a way to have the other cells overflow them, and on which side ? that would depend on the position of the inline-table in the line, and whether there are other such ruby around...  And I don't want to have to add such kludge if wordwise (not in any specs) in KOReader would become popular :)

**Comment by Frenzie** [comment]
> It's a bit tedious to get right and not ugly :)

Yup, hence why I called it quick and dirty. I didn't intend it as anything more than a quick demo of how one could approach taht.

> And I don't want to have to add such kludge if wordwise (not in any specs) in KOReader would become popular :)

Agreed.

File: frontend/ui/data/css_tweaks.lua [hunk 1]
```diff
@@ -301,6 +301,14 @@ h1, h2, h3, h4, h5, h6 { hyphens: none !important; }
 * { font-variant: no-common-ligatures; }
                 ]],
             },
+            {
+                id = "ruby_inline";
+                title = _("Render <ruby> content inline"),
+                description = _("Disable handling of <ruby> tags and have them rendered inline."),
```
Comment (Frenzie) [inline frontend/ui/data/css_tweaks.lua:307]: I just realized "have them rendered inline" seems a bit wordy. Maybe like this?
```suggestion
                description = _("Disable handling of <ruby> tags and render them inline."),
```
